docs/reference/nuctl/nuctl.md

```diff
@@ -0,0 +1,83 @@
+# Nuctl
+
+## About
+
+Nuctl is Nuclio command-line interface that provide all nuclio-features from your terminal
+
+## Download
+
+To install nuctl all you need is to simply visit Nuclio [releases page](https://github.com/nuclio/nuclio/releases)
+And download the matched binary to your platform (e.g.: `darwin` if you have `macOS`)
+
+If you are adventurous enough, you can use the following snippet
+```bash
+curl -s https://api.github.com/repos/nuclio/nuclio/releases/latest \
+			| grep -i "browser_download_url.*nuctl.*$(uname)" \
+			| cut -d : -f 2,3 \
+			| tr -d \" \
+			| wget -O nuctl -qi - && chmod +x nuctl
+```
+
+## Use
+
+Once downloaded, an informative help section is available for each command using `nuctl --help`
+
+## Running platform
+
+Nuctl will automatically identify its running platform, whether it is Docker or Kubernetes.
+
+To ensure you run Nuctl against a specific platform, use `--platform kube` for Kubernetes or `--platform local` for docker
+
+### Docker
+
+Ensure your Docker daemon is running, run  `docker version` with the same user that will execute Nuctl commands
+
+An example of function deployment using nuctl:
+
+```bash
+nuctl deploy helloworld \
+    --path https://raw.githubusercontent.com/nuclio/nuclio/master/hack/examples/golang/helloworld/helloworld.go
+```
+
+A function would be deployed, base on the source code given by the url and would be run as a docker container
+
+> NOTE: You may change the path to a local file, directory or a compressed zip
```

Comment:
This can be very misleading. User can't just do that to any local file. He has to prepare a valid nuclio function dir / code / archive